support/BRFileService.c

```diff
@@ -103,18 +177,45 @@ fileServiceEntityTypeAddHandler (BRFileServiceEntityType *entityType,
 ///
 ///
 struct BRFileServiceRecord {
-    const char *pathToType;
+    char    *sdbPath;
+    sqlite3 *sdb;
+    sqlite3_stmt *sdbInsertStmt;
+    sqlite3_stmt *sdbSelectStmt;
+    sqlite3_stmt *sdbSelectAllStmt;
+    sqlite3_stmt *sdbUpdateStmt;
+    sqlite3_stmt *sdbDeleteStmt;
+    sqlite3_stmt *sdbDeleteAllTypeStmt;
+    sqlite3_stmt *sdbDeleteAllStmt;
+
+    char *currency;
+    char *network;
+
+    pthread_mutex_t lock;
+    
     BRArrayOf(BRFileServiceEntityType) entityTypes;
     BRFileServiceContext context;
     BRFileServiceErrorHandler handler;
 };
 
+static BRFileService
+fileServiceCreateReturnError (BRFileService fs,
+                              int releaseLock,
+                              BRFileServiceError error) {
+    // Nothing with 'error' at this point; a placeholder for now.
+    if (releaseLock) pthread_mutex_unlock (&fs->lock);
+    fileServiceRelease (fs);
+    return NULL;
+}
+
 extern BRFileService
 fileServiceCreate (const char *basePath,
                    const char *currency,
                    const char *network,
                    BRFileServiceContext context,
                    BRFileServiceErrorHandler handler) {
+    if (NULL == currency || 0 == strlen(currency)) return NULL;
```

Comment:
Do we want to throw a check on basePath in here too?

Reply:
Okay.